### DISCUSSION TITLE
Why is the Raycast extension removed?

### DISCUSSION THREAD

**Discussion by Broderick-Westrope** [massCodeIO/massCode#399] [Q&A] [answered]
Hey, I was wondering why the link in the README to the Raycast extension was broken, so I did a search on Raycast only to find the extension didn't exist. After some Googling, I found [this extension](https://www.raycast.com/antonreshetov/masscode-removed) which was the README link with `-removed` at the end. Is there something wrong with the Raycast Extension? 

Apologies if this has been covered already, I couldn't find a discussion or issue on it. If the extension is not faulty it may be worth looking into fixing this misleading URL.

**Comment by antonreshetov** [comment, marked as answer]
I removed the extension because Raycast's policy towards authors seemed strange to me. First, it's necessary to keep your code in someone else's repository, but you can live with that, second, what caused me the most confusion is that anyone can make changes to the code, make a PR and it will be merged, whether the author wants it or not.

https://github.com/raycast/extensions/pull/5820

**Reply by Broderick-Westrope** [reply]
Makes sense, thanks for responding :)

**Reply by jasonshanks** [reply]
This is a real shame.  I was quite fond of MassCode and its overall open-source and json stored approach to code snippets. Unfortunately pulling the extension was the main reason I moved on, as Raycast is WAY more integrated into everything i do on my Mac.

I would urge you to reconsider. I realise it's a different approach to open-source than Github's but it's a vibrant and supportive developer community over there, in the 1.5 years I've participated in their Slack space and ecosystem I haven't heard of any developers getting upset of anyone trampling over their extensions. If anything it helps keep things that people may be relying on from languishing into irrelevance. Like, do you really have time to maintain your owns app and this extension well all the time?

Fun fact, it's actually how they run their own code base for Raycast internally. No PRs! Their small team iterates fast and gets new features out by not getting bogged down in too many PRs (obviously wouldn't work so well in a larger team but they have developed a lot of trust and confidence between them).

Meanwhile, I've moved on to a new Markdown/YAML file/folder based extension built directly out of Raycast alone, that I've been helping iterate some of the basic functionality with the main dev: https://www.raycast.com/astronight/snippetsurfer

**Reply by antonreshetov** [reply]
@jasonshanks Shame? Are you sure you know the meaning of this word?

**Reply by jasonshanks** [reply]
Hey @antonreshetov Just a longterm vernacular of my generation sorry if you took it as the other meaning. I meant as definition #2:
> 2 [in singular] a regrettable or unfortunate situation or action: what a shame Ellie won't be here | it is a shame that they are not better known.

I was going to use a 'pity' instead but you raise a good point, these both seem like insufferably christian english words don't they! 😆

Let's just leave it as unfortunate? Oh gosh, it doesn't matter. I *love* your work! Really miss the extension is just the gist of it. 😢

From a business perspective they also have a rapidly growing user base and I've seen many moving on from Alfred to Raycast (as I did a couple years ago), so a lot of opportunity for others to discover your apps as well!

**Reply by jasonshanks** [reply]
I was also just considering your other point of keeping your code in another's repository, I get that. You could always have it in yours as well of course.

But I also get *why* they've chosen this path so that the: Store experience is uniform and seamless for users novice to advanced, of even say enterprise that need to run a script and deploy a bundle of extensions at once.

In contrast I always felt this was a major failing of Alfred as trying to discover and install all the third party workflows is a **Hot Mess**, that lead to alternative repositories like Packal.org or multiple variations on people attempts to keep installed workflows updated, but not others. Alfred developers completely dropped the ball on this.

And don't get me started on Vim! 🤣 I live in my Neovim editor, but sheesh, there's like a dozen different ways to package manage (LazyVim's the best 😉). But that's kind of the cowboy vibe one expects there.
